File: cpp/alerts/torrent_delete_failed.cpp
#include "./torrent_delete_failed.h"
#include "cpp/error.h"

namespace ltrs {

Error torrent_delete_failed_alert_get_error(lt::torrent_delete_failed_alert* alert) {
    return error_code_to_error(alert->error);
}

InfoHashCpp torrent_delete_failed_alert_get_info_hashes(lt::torrent_delete_failed_alert* alert) {
    return info_hash_t_to_info_hash_cpp(alert->info_hashes);
}

} // namespace ltrs

File: src/ffi/alerts/torrent_delete_failed.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type torrent_delete_failed_alert = crate::ffi::ffi::torrent_delete_failed_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/torrent_delete_failed.h");
        type Error = crate::ffi::error::ffi::Error;
        type InfoHashCpp = crate::ffi::ffi::InfoHashCpp;

        unsafe fn torrent_delete_failed_alert_get_error(
            alert: *mut torrent_delete_failed_alert,
        ) -> Error;
        unsafe fn torrent_delete_failed_alert_get_info_hashes(
            alert: *mut torrent_delete_failed_alert,
        ) -> InfoHashCpp;
    }
}
